src/waveform/renderers/waveformwidgetrenderer.cpp

```diff
@@ -242,6 +225,72 @@ void WaveformWidgetRenderer::draw(QPainter* painter, QPaintEvent* event) {
     //qDebug() << "draw() end" << timer.restart().formatNanosWithUnit();
 }
 
+void WaveformWidgetRenderer::drawPlayPosmarker(QPainter* painter) {
+    const int lineX = m_width * m_playMarkerPosition;
+    const int lineY = m_height * m_playMarkerPosition;
+
+    // draw dim outlines to increase playpos/waveform contrast
+    painter->setOpacity(0.5);
+    painter->setPen(m_colors.getBgColor());
+    QBrush bgFill = m_colors.getBgColor();
+    if (m_orientation == Qt::Horizontal) {
+        // lines next to playpos
+        // Note: don't draw lines where they would overlap the triangles,
+        // otherwise both translucent strokes add up to a darker tone.
+        painter->drawLine(lineX + 1, 4, lineX + 1, m_height);
+        painter->drawLine(lineX - 1, 4, lineX - 1, m_height);
+
+        // triangle at top edge
+        // Increase line/waveform contrast
+        painter->setOpacity(0.8);
+        QPointF t0 = QPointF(lineX - 5, 0);
+        QPointF t1 = QPointF(lineX + 5, 0);
+        QPointF t2 = QPointF(lineX, 6);
+        drawTriangle(painter, bgFill, t0, t1, t2);
+    } else { // vertical waveforms
+        painter->drawLine(4, lineY + 1, m_width, lineY + 1);
+        painter->drawLine(4, lineY - 1, m_width, lineY - 1);
+        // triangle at left edge
+        painter->setOpacity(0.8);
+        QPointF l0 = QPointF(0, lineY - 5.01);
+        QPointF l1 = QPointF(0, lineY + 4.99);
+        QPointF l2 = QPointF(6, lineY);
+        drawTriangle(painter, bgFill, l0, l1, l2);
+    }
+
+    // draw colored play position indicators
+    painter->setOpacity(1.0);
+    painter->setPen(m_colors.getPlayPosColor());
+    QBrush fgFill = m_colors.getPlayPosColor();
+    if (m_orientation == Qt::Horizontal) {
+        // play position line
+        painter->drawLine(lineX, 0, lineX, m_height);
+        // triangle at top edge
+        QPointF t0 = QPointF(lineX - 4, 0);
+        QPointF t1 = QPointF(lineX + 4, 0);
+        QPointF t2 = QPointF(lineX, 5);
+        drawTriangle(painter, fgFill, t0, t1, t2);
+    } else {
+    // vertical waveforms
+        painter->drawLine(0, lineY, m_width, lineY);
+        // triangle at left edge
+        QPointF l0 = QPointF(0, lineY - 4.01);
+        QPointF l1 = QPointF(0, lineY + 4);
+        QPointF l2 = QPointF(5, lineY);
+        drawTriangle(painter, fgFill, l0, l1, l2);
+    }
+}
+
+void WaveformWidgetRenderer::drawTriangle(QPainter* painter, QBrush fillColor, QPointF p0, QPointF p1, QPointF p2) {
+    QPainterPath triangle;
```

Comment:
Had to apply this patch, otherwise it would refuse build:

```udiff
diff --git a/src/waveform/renderers/waveformwidgetrenderer.cpp b/src/waveform/renderers/waveformwidgetrenderer.cpp
index 074f87a345..b8ab58e905 100644
--- a/src/waveform/renderers/waveformwidgetrenderer.cpp
+++ b/src/waveform/renderers/waveformwidgetrenderer.cpp
@@ -1,4 +1,5 @@
 #include <QPainter>
+#include <QPainterPath>

 #include "waveform/renderers/waveformwidgetrenderer.h"
 #include "waveform/waveform.h"
```